runtime/doc/lsp.txt

```diff
@@ -24,88 +24,82 @@ QUICKSTART                                              *lsp-quickstart*
 Nvim provides an LSP client, but the servers are provided by third parties.
 Follow these steps to get LSP features:
 
-  1. Install the nvim-lspconfig plugin.  It provides common configuration for
-     various servers so you can get started quickly.
-     https://github.com/neovim/nvim-lspconfig
-  2. Install a language server. A list of language servers can be found here:
+  1. Install language servers using your package manager or by
+     following the upstream installation instruction.
+
+     A list of language servers is available at:
+
      https://microsoft.github.io/language-server-protocol/implementors/servers/
-     See individual server documentation for installation instructions.
-  3. Add `lua require('lspconfig').xx.setup{…}` to your init.vim, where "xx" is
-     the name of the relevant config. See the nvim-lspconfig README for details.
-     NOTE: Make sure to restart nvim after installing and configuring.
-  4. Check that an LSP client has attached to the current buffer:  >
 
-      :lua print(vim.inspect(vim.lsp.buf_get_clients()))
+  2. Configure the LSP client per language server.
+     A minimal example:
+>
+     vim.lsp.start({
+        name = 'my-server-name',
+        cmd = {'name-of-language-server-executable'},
+        root_dir = vim.fs.dirname(vim.fs.find({'setup.py', 'pyproject.toml'}, { upward = true })[1]),
```

Comment:
Having it be a function would also allow potentially running it in async mode, making things like shelling out to locate an appropriate root dir less disruptive to the UX

Reply:
> should this be a function so that it's eval'd dynamically?

I'm not sure if it is needed, given that `vim.lsp.start` will likely be called within a FileType event most of the time, so you already get some lazy evaluation due to that.

If we end up adding `lsp.config` too, that may change things.

> Having it be a function would also allow potentially running it in async mode, making things like shelling out to locate an appropriate root dir less disruptive to the UX

Only if the function supported a callback.